src/channel-qvm.lisp

```diff
@@ -0,0 +1,204 @@
+;;;; channel-qvm.lisp
+;;;;
+;;;; Authors: Sophia Ponte, Nikolas Tezak, Erik Davis
+
+(in-package #:qvm)
+
+;;; Implements a QVM that can interpret a rule-based noise model and
+;;; apply kraus operators to it's state.
+
+(defgeneric apply-all-kraus-maps (qvm instr kraus-maps)
+  (:documentation "Applies every kraus-map in the list KRAUS-MAPS to the state of the system."))
+
+
+(defgeneric apply-kraus-map (qvm instr kraus-ops)
+  (:documentation "Applies noise from a kraus map to the system. Randomly select a kraus operator from the kraus map KRAUS-OPS to apply to the state of the system."))
```

Comment:
Specify what you mean by "randomly." My guess is you mean "uniformly at random."